### DISCUSSION TITLE
Firmware and boot issues

### DISCUSSION THREAD

**Discussion by Shruthishettys** [vmware/packer-examples-for-vsphere#847] [Help] [answered]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

develop

### Motivation

When switching my vSphere’s VM boot mechanism from BIOS to (U)EFI, the virtual cd-rom drive (with windows 11 iso mounted) isn’t detected as a boot device.
ISO File name - [Win_Pro_11_22H2.9_64BIT[](url)](url)

Below are the files i see when using the bios.
Also, I attached answer file which i am using for efi

[autounattend.txt](https://github.com/vmware-samples/packer-examples-for-vsphere/files/14378564/autounattend.txt)

**Comment by tenthirtyam** [comment, marked as answer]
I have coverted this to a **Help** discussion.

1. Only UEFI is used in this project for the firmware.
2. Windows 11 requires a TPM and [use of vTPM requires UEFI](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.security.doc/GUID-6F811A7A-D58B-47B4-84B4-73391D55C268.html#vsphere-requirements-for-vtpms-1).

**Comment by shettys-ops** [comment]
I just want to change the firmware to efi-secure so that secureboot is enabled. I have bypass tpm in my answer file..
Just that the iso file which works when the firmware is set to bios fails to detect the boot loader and drives when I change it to efi-secure

**Reply by tenthirtyam** [reply]
Sorry, but I'm not following you.

**Reply by tenthirtyam** [reply]
What are you creating issues in two projects - here and the plugin repository?

https://github.com/hashicorp/packer-plugin-vsphere/issues/366?notification_referrer_id=NT_kwDOAHaU47I5NTc1OTgyMTMxOjc3NzEzNjM

**Comment by shettys-ops** [comment]
Sorry for the confusions
This should be under vsphere-iso packer..
As I m trying to create the Packer templates for Windows 11 with secure boot enabled..

**Reply by tenthirtyam** [reply]
This project's examples provide this.

**Reply by Shruthishettys** [reply]
Yes, I have reviewed those examples, thank you. 
Could you please assist me in understanding whether it is essential to utilize 'cd_files' for uploading the Windows ISO and 'cd_content' for the answer file?
In my scenario, I am using 'floppy_files' and 'floppy_content' as follows, as 'cd_files' necessitates the installation of a command-line tool for managing ISO creation.
 However, when I execute this, it generates an error stating 'EFI Floppy... No compatible bootloader found.' Could you provide guidance on this issue?"

Code reference:
iso_paths             = ["[${var.common_iso_datastore}] ${var.iso_path}/${var.win11_iso_file}", "[${var.common_iso_datastore}] ${var.iso_path}/${var.vmware_iso_file}"]
  boot_command         = ["<spacebar><spacebar>"]
  floppy_files         = ["windows-init.ps1","windows-vmtools.ps1"]
  floppy_content = {
    "autounattend.xml" = templatefile("autounattend.pkrtpl.hcl", {
      winrm_password       = var.winrm_password
    })
  }
